packages/cli/src/notices/notices_files.ts

```diff
@@ -0,0 +1,56 @@
+import { typedConfigurationFileFactory } from '@aws-amplify/platform-core';
+import { z } from 'zod';
+import { noticesManifestSchema } from '@aws-amplify/cli-core';
+
+const acknowledgementFileSchema = z.object({
+  projectAcknowledgements: z.array(
+    z.object({
+      projectName: z.string(),
+      noticeId: z.string(),
+      acknowledgedAt: z.number(),
+    }),
+  ),
```

Comment:
do we need a separate file just for acknowledgements? Can't we use `notices_printing_tracker`?

Reply:
I was going back and forth between 1 file and multiple files.

The reason I extracted ack file is that this is most impactful resource should corruption happen. If it's broken then notices will come back to customer.

I was debating whether to unify printing and cache into `notices_metadata.json` file. Loosing these two has minimal impact.

@Amplifiyer wdyt?

Reply:
one more thing I was looking at is write frequencly.

ack write is rare event. needs customer action

cache or tracker writes are performed more often.

therefore separating write patterns decreases probablity of ack file corruption. (for example if node is killed mid write).

Reply:
sounds good! We can go with two files!